### ISSUE TITLE
Normalizing flows tutorial uses outdated SELFIES methods

### ISSUE THREAD

**Issue by seyonechithrananda** [deepchem/deepchem#2530]
## 🐛 Bug

The existing normalizing flows tutorial uses outdated SELFIES methods to convert from strings to one-hot embeddings and back. I've been working on a fix to move to the newer methods (which take in different arguments) but am running into some issues. 

Namely, 'multiple_selfies_to_hot' needs to be converted to use 'batch_selfies_to_flat_hot', and 'multiple_hot_to_selfies' needs to be converted to use 'batch_flat_hot_to_selfies'. You can find the utils file with the updated methods in the SELFIES repo [here](https://github.com/aspuru-guzik-group/selfies/blob/master/selfies/utils.py#L242).

I've gotten 'batch_selfies_to_flat_hot()' to work as a replacement in the tutorial, but upon trying to convert the one-hot encodings back to SELFIES strings I run into the following error:

## To Reproduce

Steps to reproduce the behaviour:

Just need to run the normalizing flows tutorial in Colab to reproduce the error with encoding/decoding one-hot encodings.

## Expected behaviour

I've gotten 'batch_selfies_to_flat_hot()' to work as a replacement in the tutorial, but upon trying to convert the one-hot encodings back to SELFIES strings I run into the following error:
![image](https://user-images.githubusercontent.com/46096704/120381792-e41d5800-c2f0-11eb-9c4e-dfc02f9aedbd.png)

## Environment

* OS: Windows 10
* Python version: 3.7.10
* DeepChem version: Latest stable version (2.5.0)
* RDKit version (optional): latest version alongside fresh DeepChem install
* TensorFlow version (optional): 2.5.0 (latest in pip install)
* PyTorch version (optional): n/a 
* Any other relevant information: n/a

## Additional context

Once I get the secondary method to work, I'll put up a PR to fix the tutorial. Would love any ideas as to what may be the issue here. 

My suspicion is that this has to do with the methods used in the tutorial to clip and floor the generated sample. Some vectors in 'mols_list' have more than a single 1 (each must be within the bounds of {0,1} which it fulfils). Maybe using argmax and then converting to a flattened onehot embedding may work instead? 

@ncfrey @rbharath would love your thoughts on this! I can also share my updated notebook if that helps.

**Comment by seyonechithrananda** [comment]
Edit: #2320 introduces a potential fix but was never merged in. I will try this out and see if it works with the latest SELFIES version.

**Comment by schithranandanurix** [comment]
Update - running into some problems with #2320 's fix - namely that the validity goes down to 0.80%. Trying out the batch_flat methods instead but with a different set of transformations then clipping + flooring to convert the generated sample distribution to {0,1]
